jina/types/arrays/memmap.py

```diff
@@ -65,6 +66,10 @@ def __init__(self, path: str, key_length: int = 36):
         self._key_length = key_length
         self._load_header_body()
 
+    @cached_property
+    def _mmap(self):
+        return mmap.mmap(self._body_fileno, length=0, prot=mmap.PROT_READ)
```

Comment:
Are we sure that opening the file only once makes it work when `append` and `update` operations are done?

Reply:
I do not think this will work when the `file` size is increased.


```
class mmap.mmap(fileno, length, flags=MAP_SHARED, prot=PROT_WRITE|PROT_READ, access=ACCESS_DEFAULT[, offset])
(Unix version) Maps length bytes from the file specified by the file descriptor fileno, and returns a mmap object. If length is 0, the maximum length of the map will be the current size of the file when mmap is called.
```

Reply:
we want to invalidate the mmap?

Reply:
That's true, converting to draft.